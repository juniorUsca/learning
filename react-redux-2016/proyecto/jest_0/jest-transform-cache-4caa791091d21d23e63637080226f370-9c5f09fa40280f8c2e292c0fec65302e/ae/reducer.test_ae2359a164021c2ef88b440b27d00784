9556de7b6288a330546a19dfcb7a47c3
'use strict';var _reducer = require('./reducer');var _reducer2 = _interopRequireDefault(_reducer);
var _actions = require('./actions');var _actions2 = _interopRequireDefault(_actions);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// vamos a probar la acción `SET_COMMENTS` cuando se envía al store
test('reducer - SET_COMMENTS', () => {
  expect(
  (0, _reducer2.default)(
  undefined,
  _actions2.default.setComments([{ id: 1 }, { id: 2 }]))).

  toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlZHVjZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJ0ZXN0IiwiZXhwZWN0IiwidW5kZWZpbmVkIiwic2V0Q29tbWVudHMiLCJpZCIsInRvTWF0Y2hTbmFwc2hvdCJdLCJtYXBwaW5ncyI6ImFBQUEsb0M7QUFDQSxvQzs7QUFFQTtBQUNBQSxLQUFLLHdCQUFMLEVBQStCLE1BQU07QUFDbkNDO0FBQ0U7QUFDQ0MsV0FERDtBQUVFLG9CQUFRQyxXQUFSLENBQW9CLENBQUMsRUFBRUMsSUFBSSxDQUFOLEVBQUQsRUFBWSxFQUFFQSxJQUFJLENBQU4sRUFBWixDQUFwQixDQUZGLENBREY7O0FBS0VDLGlCQUxGO0FBTUQsQ0FQRCIsImZpbGUiOiJyZWR1Y2VyLnRlc3QuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgcmVkdWNlciBmcm9tICcuL3JlZHVjZXInO1xuaW1wb3J0IGFjdGlvbnMgZnJvbSAnLi9hY3Rpb25zJztcblxuLy8gdmFtb3MgYSBwcm9iYXIgbGEgYWNjacOzbiBgU0VUX0NPTU1FTlRTYCBjdWFuZG8gc2UgZW52w61hIGFsIHN0b3JlXG50ZXN0KCdyZWR1Y2VyIC0gU0VUX0NPTU1FTlRTJywgKCkgPT4ge1xuICBleHBlY3QoXG4gICAgcmVkdWNlcihcbiAgICAgdW5kZWZpbmVkLFxuICAgICAgYWN0aW9ucy5zZXRDb21tZW50cyhbeyBpZDogMSB9LCB7IGlkOiAyIH1dKVxuICAgIClcbiAgKS50b01hdGNoU25hcHNob3QoKTtcbn0pO1xuIl19