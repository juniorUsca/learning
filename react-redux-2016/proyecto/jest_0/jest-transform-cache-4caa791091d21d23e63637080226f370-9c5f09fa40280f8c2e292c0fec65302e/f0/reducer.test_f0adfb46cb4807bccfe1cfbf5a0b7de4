5706e6b690005bd2650d50c4d4937e7d
"use strict";
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsImZpbGUiOiJyZWR1Y2VyLnRlc3QuanMiLCJzb3VyY2VzQ29udGVudCI6W119