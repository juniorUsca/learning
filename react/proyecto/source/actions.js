import api from './api'

// action creators
function setPosts(posts) {
  return {
    type: 'SET_POSTS',
    payload: posts,
  };
}

function setPost(post) {
  return {
    type: 'SET_POST',
    payload: post,
  };
}

function setComments(comments) {
  return {
    type: 'SET_COMMENTS',
    payload: comments,
  };
}

function setUser(user) {
  return {
    type: 'SET_USER',
    payload: user,
  };
}

function postsNextPage() {
  return async (dispatch, getState) => {
    console.log('accion asincrona');
    const state = getState()
    // const currentPage = state.posts.page
    const currentPage = state.get('posts').get('page')
    const posts = await api.posts.getList(currentPage)
    dispatch(
      setPosts(posts),
    )
    return posts;
  };
}

function loadCommentsForPost(postId) {
  return async (dispatch) => {
    const comments = await api.posts.getComments(postId)
    dispatch(
      setComments(comments),
    )
    return comments;
  };
}

function loadUser(userId) {
  return async (dispatch) => {
    const user = await api.users.getSingle(userId)
    dispatch(
      setUser(user),
    )
    return user;
  };
}

function loadUserPosts(userId) {
  return async (dispatch) => {
    const posts = await api.users.getPosts(userId)
    dispatch(
      setPosts(posts),
    )
    return posts;
  };
}

function loadPost(id) {
  return async (dispatch) => {
    const post = await api.posts.getSingle(id)
    dispatch(
      setPost(post),
    )
    return post;
  };
}

export default {
  setPosts,
  setComments,
  setUser,
  postsNextPage,
  loadCommentsForPost,
  loadUser,
  loadUserPosts,
  loadPost,
};
