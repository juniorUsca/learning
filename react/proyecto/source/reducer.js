import { combineReducers } from 'redux'

const initialState = {
  posts: {
    page: 1,
    entities: [],
  },
  comments: [],
  users: [],
};

/* const action = {
  type: 'SET_POSTS',
  payload: {}, // data
  // meta: {}, // informacion extra
  // error: true, // si esta accion tiene error
} */

// tiene solo informacion de la pagina actual
function postsPageReducer(state = initialState.posts.page, action = {}) {
  switch (action.type) {
    case 'SET_POSTS':
      return state + 1
    default:
      return state
  }
}

function postsEntitiesReducer(state = initialState.posts.entities, action = {}) {
  switch (action.type) {
    case 'SET_POSTS':
      return state.concat(action.payload)
    default:
      return state
  }
}

// se encarga de combinar los reducers individules
const postsReducer = combineReducers({
  page: postsPageReducer,
  entities: postsEntitiesReducer,
})


function commentsReducer(state = initialState.comments, action = {}) {
  switch (action.type) {
    case 'SET_COMMENTS':
      console.log('----------------------', state)
      return state.concat(action.payload)
    default:
      return state
  }
}

function usersReducer(state = initialState.users, action = {}) {
  switch (action.type) {
    case 'SET_USER':
      console.log('----------------------', state)
      return Object.assign({}, state, {
        [action.payload.id]: action.payload,
      })
    default:
      return state
  }
}

const reducer = combineReducers({
  posts: postsReducer,
  comments: commentsReducer,
  users: usersReducer,
})

// function reducer(state = initialState, action = {}) {
//   switch (action.type) {
//     case 'SET_POSTS':
//       return Object.assign({}, state, {
//         posts: Object.assign({}, state.posts, {
//           entities: state.posts.entities.concat(action.payload),
//           page: state.posts.page + 1,
//         }),
//       });
//     default:
//       return state;
//   }
// }

export default reducer;
