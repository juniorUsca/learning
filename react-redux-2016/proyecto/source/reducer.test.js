import reducer from './reducer';
import actions from './actions';

// vamos a probar la acción `SET_COMMENTS` cuando se envía al store
test('reducer - SET_COMMENTS', () => {
  expect(
    reducer(
     undefined,
      actions.setComments([{ id: 1 }, { id: 2 }])
    )
  ).toMatchSnapshot();
});
