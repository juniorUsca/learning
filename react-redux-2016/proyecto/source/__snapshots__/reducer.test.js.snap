// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer - SET_COMMENTS 1`] = `
Immutable.Map {
  "posts": Immutable.Map {
    "page": 1,
    "entities": Immutable.Map {},
  },
  "comments": Immutable.Map {
    1: Immutable.Map {
      "id": 1,
    },
    2: Immutable.Map {
      "id": 2,
    },
  },
  "users": Immutable.Map {},
}
`;
